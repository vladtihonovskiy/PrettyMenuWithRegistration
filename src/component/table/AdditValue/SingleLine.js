import React, { Component } from 'react';


import { connect}  from 'react-redux';
import { Input } from 'muicss/react';
class SingleLine extends Component{
    constructor(props, context) {
        super(props, context);
        {
            this.state={
                clear:false,
                valid:true,
                validClass:'blue',
                inputValue:'',

            }
        }
    }
    componentWillMount(){
        if(this.props.item==='phone'||this.props.item==='name') {
            this.setState({
                validClass: 'red',
                valid: false
            })
        }
    }
    onChangeEmail=(e)=>{
        this.setState({inputValue: e.value});
    }
    findSaveValue=(getValueHeader)=> {
            if(getValueHeader!==undefined) {
                if (typeof (getValueHeader) === 'object') {
                    return getValueHeader.length;
                }
                else {
                    return getValueHeader.toString();
                }
            }else{
                return '';
            }
    }
    validateName=({target})=>{
        this.onChangeEmail(target);

        this.props.mainMass===undefined ? this.props.onChangeValue(target,-1):this.props.onChangeValue(target,1);
        if(target.value==='')
        {
            if(this.props.item==='phone'||this.props.item==='name')
            {
                this.setState({
                    validClass:'red',
                    valid:false
                })
            }else {
                this.setState({
                    validClass: 'blue',
                    valid: true
                })
            }
        }else{
            if(this.props.item=='phone')
            {
                let numbervalid=/^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/im

                let valid = numbervalid.test(target.value);
                if (valid){

                    if(target.value.length===13){
                        this.setState({
                            validClass:'blue',
                            valid:true
                        })
                    }else{
                        this.setState({
                            validClass:'red',
                            valid:false
                        })
                    }

                }else{
                    this.setState({
                        validClass:'red',
                        valid:false
                    })
                }
                // this.refs.someName.value===''
            }else if(this.props.item=='name')
            {
                let nameValidate= /^[A-Za-z\s]+$/;
                let valid = nameValidate.test(target.value);
                if (valid){

                    if(target.value.length>=4){
                        this.setState({
                            validClass:'blue',
                            valid:true
                        })
                    }else{
                        this.setState({
                            validClass:'red',
                            valid:false
                        })
                    }

                }else{
                    this.setState({
                        validClass:'red',
                        valid:false
                    })
                }
            }else{
                this.setState({
                    validClass:'blue',
                    valid:true
                })
            }

        }

    }
    // componentWillReceiveProps(nextProps){
    //     if(nextProps.clear!==this.state.clear){
    //         this.refs.someName.value = '';
    //         this.setState({
    //             clear:!this.state.clear
    //         })
    //     }
    // }

    render() {

        return (
            this.props.lineId === -1 ?
                <Input onChange={this.validateName} name={this.props.item}
                       defaultValue={this.props.item === 'phone' ? '+380' : ''} ref={this.state.inputValue}
                       className={`form-control ${this.state.validClass}`} key={this.props.item + 'Form'} type="text"
                       placeholder={this.props.item}/> :
                <Input onChange={this.validateName} name={this.props.item}
                       defaultValue={this.findSaveValue(this.props.state.SelfoneData[this.props.lineId][this.props.item])} ref={(input)=>{this.InputValue=input;}}
                       className="form-control" key={this.props.item + 'Form'} type="text"
                       placeholder={this.props.item}/>
        )
    }
}
export default connect(
    state => ({
        state:state.tableValue
    }),
    dispatch => ({
        UserLoginChek: () => {

            dispatch({ type: 'LoginUser' });
        },
        setHeader: (headers)=>{
            dispatch({ type: 'HEADER_CREATE',headers });
        }
    })
)(SingleLine);


