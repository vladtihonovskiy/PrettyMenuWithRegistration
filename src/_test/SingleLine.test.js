import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SingleLine from '../../src/component/table/AdditValue/SingleLine';
import { Provider } from 'react-redux';
import reducer from '../../src/redusers/index';
import {createStore,applyMiddleware} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../../src/sagas'
const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    reducer,
    composeWithDevTools(applyMiddleware(sagaMiddleware))
)
sagaMiddleware.run(rootSaga);




describe('SingleLine', () => {
    const mockFn = jest.fn();
    it('should be defined', () => {
        expect(SingleLine).toBeDefined();
    });
     it('should render correctly', () => {
        const tree = shallow(
            <Provider store={store}>
                <SingleLine lineId={8} handleClick={mockFn} />
            </Provider>
        );
        expect(tree).toMatchSnapshot();
     });
    it('should have a button value', () => {
        const wrapper = shallow(   <Provider store={store}>
            <SingleLine lineId={8} name='button test' handleClick={mockFn} />
        </Provider>);
        expect(wrapper.props().name).toEqual('button test');
        // expect(tree.find('.button').node.props.value).toEqual('button test');
    });
    // it('should have a click function', () => {
    //     const tree = shallow(
    //         <SingleLine name='button test' handleClick={mockFn} />
    //     );
    //     tree.simulate('click');
    //     expect(mockFn).toHaveBeenCalled();
    // });
});