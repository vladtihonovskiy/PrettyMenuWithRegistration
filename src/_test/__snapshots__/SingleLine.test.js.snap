// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleLine should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(SingleLine)
    handleClick={[Function]}
    lineId={8}
/>,
  "nodes": Array [
    <Connect(SingleLine)
      handleClick={[Function]}
      lineId={8}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "dispatch": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                        Symbol(observable): [Function],
                      }
        }
>
        <Connect(SingleLine)
                handleClick={[Function]}
                lineId={8}
        />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(SingleLine)
            handleClick={[Function]}
            lineId={8}
/>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(SingleLine)
          handleClick={[Function]}
          lineId={8}
/>,
        "_debugID": 2,
        "_renderedOutput": <Connect(SingleLine)
          handleClick={[Function]}
          lineId={8}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <Connect(SingleLine)
        handleClick={[Function]}
        lineId={8}
    />
</Provider>,
}
`;
